cs/src/core/Allocator/WorkQueueLIFO.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FASTER.core
+{
+    /// <summary>
+    /// Shared work queue that ensures one worker at any given time. Uses LIFO ordering of work.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    class WorkQueueLIFO<T>
```

Comment:
looks like we can simplify and separate the counting semantics from turn processing on/off. Sent you a proposal over IM.

Reply:
This code was needed to retain robustness in the face of varying atomicity guarantees of the structure (e.g., `ConcurrentStack`) used for storing the work.